### ISSUE TITLE
duplicate drop-downs when using django-autocomplete-light

### ISSUE THREAD

**Issue by frnmst** [farridav/django-jazzmin#210]
Hi,

I am using [django-autocomplete-light](https://github.com/yourlabs/django-autocomplete-light) to display foreign key drop-down menus. This is what I get when using django-jazzmin:

![a](https://user-images.githubusercontent.com/8964425/98453289-d25b4400-2157-11eb-9ab2-291089e426ad.png)

The menu on the left is from jango-autocomplete-light.

Standard admin:

![b](https://user-images.githubusercontent.com/8964425/98453294-e141f680-2157-11eb-83a8-3afb2b26a278.png)

Any ideas to hide django-jazzmin's menus when django-autocomplete-light menus are present?

Thank you

**Comment by farridav** [comment]
I think we fix this by ignoring the select that autocomplete light is using in main.js

On a side note, unless your using autocomplete light for something advanced, or using a very old version of Django, auto complete on foreign key fields is available from Django https://docs.djangoproject.com/en/2.0/ref/contrib/admin/#django.contrib.admin.ModelAdmin.autocomplete_fields

**Comment by frnmst** [comment]
Thanks, I didn't know about ModelAdmin.autocomplete_fields! I'll give it a go.

**Comment by djpretzel** [comment]
I had this issue as well & it was a bit confusing. I do think DAL exposes more of the underlying functionality of Select2 and can be useful in scenarios where the default Django autocomplete isn't as configurable, so compatibility would be awesome.

**Comment by farridav** [comment]
If someone is able to tell me a CSS class that is applied to the select, we can add it here https://github.com/farridav/django-jazzmin/blob/master/jazzmin/static/jazzmin/js/change_form.js#L108

And this will stop us running select2 over it..

**Comment by marcoooo** [comment]
Is there any work around available for now? I struggle to find the required css to hide the right one :-)

**Comment by desarrollosrosarinos** [comment]
I'm having the same problem. I cannot use django automcomplete because is very slow when you have many foreign keys and inlines.
Is something we can do to make it work till you fix it?
Thank you, this theme looks great.

**Comment by desarrollosrosarinos** [comment]
I fix it commenting this function in change_form.js
function applySelect2() {
        // Apply select2 to any select boxes that don't yet have it
        // and are not part of the django's empty-form inline
        //const noSelect2 = '.empty-form select, .select2-hidden-accessible, .selectfilter, .selector-available select, .selector-chosen select';
        //$('select').not(noSelect2).select2({ width: 'element' });
    }
Maybe you cloud disable this function when autocomplete light is installed.

**Comment by 6sr** [comment]
Solution provided by @desarrollosrosarinos  worked for me but it also affected the UI of other select tags. Can we do something to handle this.
Before commenting
![image](https://user-images.githubusercontent.com/23514272/153060616-7a0ad08c-470f-42d4-a3f4-5236b624fe51.png)
After commenting
![image](https://user-images.githubusercontent.com/23514272/153060705-66ac675a-dfb0-47cc-8ff4-cf2e45f30839.png)

**Comment by MathiasKowoll** [comment]
@6sr indeed it works but if you have an inline it still shows duplicated when using inlines, I applied you solution and instead of using dal on inlines, I used Django owns implementation

**Comment by MathiasKowoll** [comment]
I was able to fix the issue changing the following without applying previous fixes. It works with inlines too. I added the following class into the JavaScript [django-jazzmin/jazzmin/static/jazzmin/js/change_form.js](https://github.com/farridav/django-jazzmin/blob/e3f9d45183d58f78bf4c6793969490631a84681d/jazzmin/static/jazzmin/js/change_form.js#L116):
chango from 
const noSelect2 = '.empty-form select, .select2-hidden-accessible, .selectfilter, .selector-available select, .selector-chosen select,';
to
const noSelect2 = '.empty-form select, .select2-hidden-accessible, .selectfilter, .selector-available select, .selector-chosen select, .noSelect2DAL, ';

then in dal forms add the following:
attrs={'class': 'noSelect2DAL'}

Hope this helps and we can keep the functionality

**Comment by MathiasKowoll** [comment]
@farridav what do you think about the solution above?

**Comment by farridav** [comment]
Yeah, makes sense to me, if you want to get it in a Pull request, I'll get it reviewed and merged :)

I think it's high time I added some documentation around integration with other apps, in which I can mention how to apply this fix.

**Comment by MathiasKowoll** [comment]
> Yeah, makes sense to me, if you want to get it in a Pull request, I'll get it reviewed and merged :)
> 
> I think it's high time I added some documentation around integration with other apps, in which I can mention how to apply this fix.

Will do later today so it can be merged

**Comment by MathiasKowoll** [comment]
@farridav created the PR, let me know if everything is ok

**Comment by hdanh** [comment]
using latest django-jazzmin, I still get the duplicated issue, any idea how to fix it 
<img width="686" alt="image" src="https://user-images.githubusercontent.com/8652214/187373995-7c531ce8-0f3c-4c85-8cd2-fcf9dcd92e5c.png">

**Comment by farridav** [comment]
Have you tried the fix from @MathiasKowoll ?

**Comment by hdanh** [comment]
@farridav ah it works, thanks

**Comment by MathiasKowoll** [comment]
@hdanh I'm glad it worked

**Comment by Simanas** [comment]
I have submitted pull request with a beautiful fix which doesn't even require any use of additional attributes on autocomplete widgets. Please merge it in. It works seamlessly!

**Comment by MathiasKowoll** [comment]
> I have submitted pull request with a beautiful fix which doesn't even require any use of additional attributes on autocomplete widgets. Please merge it in. It works seamlessly!

Does It work on inlines too?

**Comment by Simanas** [comment]
Yes.
